netstandard/tools/targets/NETStandard.Library.Trimming.targets

```diff
@@ -0,0 +1,74 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="TrimFiles" AssemblyFile="$(NETStandardToolsTaskDirectory)NETStandard.Tools.dll" />
+
+  <Target Name="_determineTrimPackageInputs">
+    <ItemGroup Condition="'$(RootPackageReference)' != 'true'">
+      <TrimFilesRootPackages Include="@(PackageReference)"/>
+    </ItemGroup>
+
+    <!-- todo: move this stuff to NETCore.App package -->
+    <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+      <!-- Native host / CLR for netcoreapp -->
+      <TrimFilesRootFiles Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <TrimFilesRootFiles Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <TrimFilesRootFiles Include="dotnet.exe;dotnet" />
+      <TrimFilesRootFiles Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <TrimFilesRootFiles Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
+
+      <!-- treat platform packages as trimmable -->
+      <TrimmablePackages Include="$(PackageConflictPreferredPackages)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="TrimFilesOnBuild"
+          DependsOnTargets="_determineTrimPackageInputs"
+          AfterTargets="CoreCompile">
+    <ItemGroup>
+      <_trimOtherRuntimeItems Include="@(_LockFileAssemblies)" Exclude="@(_ConflictPackageFiles)" />
+    </ItemGroup>
+
+    <TrimFiles RootFiles="@(IntermediateAssembly);@(TrimFilesRootFiles)"
```

Comment:
How does the perf look like for this? I saw you are doing some caching for the lock file but you are calling TrimFiles on every build and publish. I guess it only happens if Compile happens so it is triggered only when Compile is triggered.